pkg/lsp/jsonrpc2.go

```diff
@@ -0,0 +1,52 @@
+package lsp
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// ID represents a JSON-RPC 2.0 request ID, which may be either a
+// string or number (or null, which is unsupported).
+type ID struct {
```

Comment:
This package in particular just contains LSP definitions, and has minimal dependencies. The language server isn't the only consumer of it, so it's common in go to do this sort of thing to avoid pulling in larger packages unnecessarily.

In fact it is a go proverb:
> A little copying is better than a little dependency.

https://go-proverbs.github.io/